File: src/TestJava/AgeCalculator.java
package TestJava;

import java.time.LocalDate;
import java.time.Period;
import java.util.Scanner;
import java.time.format.DateTimeFormatter;

public class AgeCalculator {
    public static void main(String[] args) {
        int Age;
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter your date of birth in YYYY-MM-DD format : ");
        String DOB = sc.next();

        LocalDate currentdate = LocalDate.now();
        LocalDate dob = LocalDate.parse(DOB, DateTimeFormatter.ISO_LOCAL_DATE);
        Age = Period.between(dob,currentdate).getYears();
        System.out.println("Your age is: "+ Age);

    }
}

File: src/KotlinSamples/AgeCalculator.kt
package KotlinSamples

import java.time.LocalDate
import java.time.Period
import java.time.format.DateTimeFormatter

fun main() {
    val ageInYears : Int
    println("Enter your date of birth in YYYY-MM-DD(2000-01-01) format: ")
    val strAge: String? = readLine()
    val currentDate  = LocalDate.now ()
    val dob = LocalDate.parse(strAge, DateTimeFormatter.ISO_LOCAL_DATE)
    ageInYears = Period.between(dob,currentDate).years
    println("Age in years: $ageInYears")
    val ageInMonths = Period.between(dob,currentDate).months
    val ageInDays = Period.between(dob,currentDate).months
    println("You are $ageInYears years $ageInMonths months $ageInDays days old")


}